### PR TITLE
feat(connector): support more parquet data types

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Since RisingWave does not have Uint data type, previously we do not support `UIntx`(`UInt8, UInt16, UInt32, UInt64`) in parquet encode, and `Float16` is also not supported.

However, user have a bad experience when they encounter an error of unsupported data type when using file_scan() or parquet file source. 

This pr add some types that can be workarounded:
```
Int8  -> Int16
UInt8  -> Int16
UInt16 -> Int32
UInt32 -> Int64
UInt64 -> Decimal
Float16 -> Float32
```
For example, if the Parquet file that the user needs to read contains a column of Uint64 type, this column should be defined as Decimal type when creating the RisingWave table.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

We need the documentation team to note these special conversions and inform us of the list of data types that we do not support.

Perhaps in https://docs.risingwave.com/integrations/sources/apache-iceberg#data-type-mapping

</details>

### CODE REVIEW

**Review by chenzl25 — APPROVED** [review]
LGTM!

File: src/connector/src/source/iceberg/parquet_file_handler.rs [hunk 1]
```diff
@@ -318,3 +320,67 @@ pub async fn get_parquet_fields(
...
+/// It handles the following special cases:
+/// - Arrow's `timestamp(_, None)` types (all four time units) match with RisingWave's `TimeStamp` type.
+/// - Arrow's `timestamp(_, Some)` matches with RisingWave's `TimeStamptz` type.
+/// - Since RisingWave does not have an `UInt` type:
+///   - Arrow's `UInt8` matches with RisingWave's `Int16`.
+///   - Arrow's `UInt16` matches with RisingWave's `Int32`.
+///   - Arrow's `UInt32` matches with RisingWave's `Int64`.
+///   - Arrow's `UInt64` matches with RisingWave's `Decimal`.
+/// - Arrow's `Float16` matches with RisingWave's `Float32`.
+fn is_data_type_match(arrow_data_type: &ArrowDateType, rw_data_type: &RwDataType) -> bool {
```
[the remark this answers is not in the document]
Comment (wcy-fdu) [reply src/connector/src/source/iceberg/parquet_file_handler.rs:334]: Sounds reasonable, `arrow -> risingwave` means it's only used in source, let's call it `is_parquet_schema_match_source_schema` or something else.

**Review by xiangjinwu — COMMENTED** [review]
Seems `int8` is missing from some parts of the test as well as implementation.

**Comment by wcy-fdu** [comment]
> Seems `int8` is missing from some parts of the test as well as implementation.

see 
```
#[test]
    fn int8() {
        let array: PrimitiveArray<i16> = I16Array::from_iter([None, Some(-128), Some(127)]);
        let arr = arrow_array::Int8Array::from(vec![None, Some(-128), Some(127)]);
        let converted: PrimitiveArray<i16> = (&arr).into();
        assert_eq!(converted, array);
    }
```

**Review by fuyufjh — APPROVED** [review]
LGTM!

**Comment by fuyufjh** [comment]
Perhaps a bit off-topic. @chenzl25 I think the iceberg test need to be enhanced to cover all data types, for both source and sink.

- Iceberg source

https://github.com/risingwavelabs/risingwave/blob/3607db5e1f5998905fe93497cfe6b0d32da43606/integration_tests/iceberg-source/python/main.py#L48-L53

- Iceberg sink

https://github.com/risingwavelabs/risingwave/blob/5065bb4d1e4ed81aa38f784ae4acb4b644c3e6a8/e2e_test/sink/iceberg_sink.slt#L5


Search for "all_types" and you will see most connectors are already covered, such as https://github.com/risingwavelabs/risingwave/blob/1eb4730cdf1602cf80ac7bb472ec7a65f044a4b8/integration_tests/nats/create_mv.sql#L1-L35

File: src/common/src/array/arrow/arrow_impl.rs [hunk 2]
```diff
@@ -505,6 +505,11 @@ pub trait FromArrow {
             Int16 => DataType::Int16,
             Int32 => DataType::Int32,
             Int64 => DataType::Int64,
+            UInt8 => DataType::Int16,
```
Comment (xiangjinwu) [inline src/common/src/array/arrow/arrow_impl.rs:508]: Do we need `Int8` here? If yes, why is it not caught by tests?

File: e2e_test/s3/file_sink.py [hunk 3]
```diff
@@ -62,6 +69,11 @@ def _table():
         sex bigint,
         mark bigint,
         test_int int,
+        test_uint8 smallint,
```
Comment (xiangjinwu) [inline e2e_test/s3/file_sink.py:72]: Do we need `test_int8` column here?

File: src/common/src/array/arrow/arrow_impl.rs (hunk 2 above)
Comment (wcy-fdu) [reply src/common/src/array/arrow/arrow_impl.rs:509]: Uncover by test.

**Comment by chenzl25** [comment]
> Perhaps a bit off-topic. @chenzl25 I think the iceberg test need to be enhanced to cover all data types, for both source and sink.

Indeed. But actually, the iceberg all types test is here. Agree to add a dedicated test.
https://github.com/risingwavelabs/risingwave/blob/34a2ad42da6a6dcb45b1260b6cecfce89d1c360c/e2e_test/iceberg/test_case/iceberg_sink_no_partition_append_only_table.slt#L12

**Pull request merged**